internal/controllers/nmc_reconciler.go

```diff
@@ -235,6 +247,7 @@ func GetContainerStatus(statuses []v1.ContainerStatus, name string) v1.Container
 
 type nmcReconcilerHelper interface {
 	GarbageCollectInUseLabels(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig) error
+	GarbageCollectWorkerPods(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig) error
```

Comment:
I think this methods belongs in the https://github.com/kubernetes-sigs/kernel-module-management/blob/946be3a4a12a417cb0d2486b510023edbc80d13f/internal/controllers/nmc_reconciler.go#L743 interface instead of the `nmcReconcilerHelper` interface.

The pod manager will probably move to its own package as a followup to https://github.com/kubernetes-sigs/kernel-module-management/pull/1010 so I think it will better to move it there so it moves with the rest of the interface to the `pod` package once merged.

Reply:
I would keep the function in the nmcReconcilerHelper interface.This is the logic of controller, and not of the pod. It will use the pod interface functions , like ListPods and DeletePods, but the decision which pods to delete should be in the controller

Reply:
I don't think there is a point of having a "pure pod package" as it is not different than what is already offered in https://pkg.go.dev/k8s.io/kubernetes/pkg/api/v1/pod.

This function is no different in its skope than any other method in https://github.com/kubernetes-sigs/kernel-module-management/blob/946be3a4a12a417cb0d2486b510023edbc80d13f/internal/controllers/nmc_reconciler.go#L743-L750 so I think it should be there.

Reply:
but those functions are not making decisions for the controller: they use the data the controller passes them to do something with the actual pods: create, create templates, delete. GarbageCollectWorkerPods needs to decide which pods to delete based on the NMC, Module etc', and this decision must belong to the controller

Reply:
You convinced me.